fieldservice_recurring/models/fsm_recurring.py

```diff
@@ -0,0 +1,237 @@
+        return self.write({'state': 'cancel'})
+
+    def _get_rruleset(self):
+        self.ensure_one()
+        ruleset = rruleset()
+        if self.state != 'progress':
+            return ruleset
+        # set next_date which is used as the rrule 'dtstart' parameter
+        next_date = datetime.now()
+        last_order = self.env['fsm.order'].search([
+            ('fsm_recurring_id', '=', self.id),
+            ('stage_id', '!=', self.env.ref(
+                'fieldservice.fsm_stage_cancelled').id)
+        ], offset=0, limit=1, order='scheduled_date_start desc')
+        if last_order:
+            next_date = fields.Datetime.from_string(
+                last_order.scheduled_date_start)
+        # set thru_date to use as rrule 'until' parameter
+        days_ahead = self.fsm_frequency_set_id.schedule_days
+        request_thru_date = datetime.now() + relativedelta(days=+days_ahead)
+        if self.end_date and (self.end_date < request_thru_date):
+            thru_date = self.end_date
+        else:
+            thru_date = request_thru_date
+        # use variables to calulate and return the rruleset object
+        ruleset = self.fsm_frequency_set_id._get_rruleset(dtstart=next_date,
+                                                          until=thru_date)
+        return ruleset
+
+    def _prepare_order_values(self, date=None):
+        self.ensure_one()
+        schedule_date = date if date else datetime.now()
+        days_early = self.fsm_frequency_set_id.buffer_early
+        earliest_date = schedule_date + relativedelta(days=-days_early)
+        return {
+            'fsm_recurring_id': self.id,
+            'customer_id': self.customer_id.id,
+            'location_id': self.location_id.id,
+            'scheduled_date_start': schedule_date,
+            'request_early': str(earliest_date),
+            'description': self.description,
+            'template_id': self.fsm_order_template_id.id,
+            'company_id': self.company_id.id,
+        }
+
+    def _create_order(self, date):
+        self.ensure_one()
+        vals = self._prepare_order_values(date)
+        return self.env['fsm.order'].create(vals)
+
+    @api.model
+    def _cron_generate_orders(self):
+        """
+        Executed by Cron task to create field service orders from any
+        recurring orders which are in progress, or to renew, and up to
+        the max orders allowed by the recurring order
+        @return {recordset} orders: all the order objects created
+        """
+        orders = self.env['fsm.order']
+        for rec in self.env['fsm.recurring'].search([
+            ('state', 'in', ('progress', 'pending'))
+        ]):
+            schedule_dates = rec._get_rruleset()
+            order_dates = []
+            for order in rec.fsm_order_ids:
+                if order.scheduled_date_start:
+                    order_dates.append(
+                        datetime.strptime(order.scheduled_date_start,
```

Comment:
Error here

 ` File "/home/odoo/build/OCA/field-service/fieldservice_recurring/models/fsm_recurring.py", line 184, in _cron_generate_orders
    DEFAULT_SERVER_DATETIME_FORMAT
ValueError: <class 'TypeError'>: "strptime() argument 1 must be str, not datetime.datetime" while evaluating
'model._cron_scheduled_task()'`

Reply:
Fixed.